plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLBackgroundParsingJob.java

```diff
@@ -354,7 +353,14 @@ public Interval runWithResult() {
             log.error(ex);
             return;
         }
-        IProgressMonitor monitor = Job.getJobManager().createProgressGroup();
+        
+        String textTitle = UIUtils.syncExec(new RunnableWithResult<String>() {
+            @Override
+            public String runWithResult() {
+                return editor.getTitle();
+            }
+        });
```

Comment:
I don't quite understand the purpose of this code. It should be as simple as

```suggestion
        String textTitle = editor.getTitle();
```

Reply:
It's executing not in the UI thread

Reply:
But this method isn't required to be called on the UI thread

Reply:
Ok, I can remove it.
Are there any references of the workbench part methods being thread-safe?